src/xwalk/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -51,6 +51,10 @@
 
 	private final static int ACCESS_FINE_LOCATION_PERMISSION_REQUEST = (int)Math.random();
 
+	private static final int DISCLOSURE_LOCATION_PERMISSION_REQUEST = 1122331;
```

Comment:
I don't think this should be the same int as the one above. It shouldn't be a problem but if both classes exist then the wrong one may catch the response. I think it should be random, like the ACCESS_FINE_LOCATION_PERMISSION_REQUEST above.

Reply:
I saw that the constant above that is assigned with a random number and looking in the internet why we would use a random number in a constant I couldn't find a good reason, but yes random numbers are used in `requestCode` variables when you need a new value each time, but this is not the case, the variable is only generated one time (`final`), so why we are doing that?

Constant are constants, and using a random value at startup time looks like a bad idea specially in Java, because you cannot use them in `switch` statements or annotations if the value set is assigned dynamically. Also there is the risk (unlikely) for collisions is two constants defined like that are caught in the `onActivityResult(..)` method that catch most events.

Reply:
Checking around the code again, I found that while in the Webview version we define the constant randomly like this:

```java
final static int ACCESS_FINE_LOCATION_PERMISSION_REQUEST = (int)(Math.random() * 1000);
```

In the XWalk version is:

```java
final static int ACCESS_FINE_LOCATION_PERMISSION_REQUEST = (int)Math.random();
```

`Math.random()` always return a `double` between 0 and 1 but never 1, so casting the result to `int` always get `0`. I guess this is a bug (but not resulting in errors for now) and not intended.

Reply:
Yeah definitely a bug - I should have fixed that when I fixed the Webview version, sorry!

I think the reason it's random is so it's different every time the app starts so you don't get a random response from a previous request. I'm happy to go with whatever the best practice is, if you can find one!

It seems like a strange API for someone used to callbacks...

Reply:
:thinking: didn't know that the app can restart and still receive a response from a request from a previous process, in that case makes sense. Next event variable caught by the same method can be defined using different random ranges to avoid collisions, eg:

```
jshell> (int)(Math.random() * 1000)
$9 ==> 180

jshell> (int)(Math.random() * 1000)
$10 ==> 739

jshell> (int)(Math.random() * 1000) + 1000    // different range
$11 ==> 1819

jshell> (int)(Math.random() * 1000) + 1000
$12 ==> 1679

jshell>
```

So going to make it random and fix the one with the random that is always 0.

Reply:
Fixed using random values,  and tested in XWalk and Webview